### PR TITLE
[SPARK-24888][CORE] spark-submit --master spark://host:port --status/kill driver-id does not give any status/update

### PR DESCRIPTION
**Pull request by devaraj-kavali**
## What changes were proposed in this pull request?

In `SparkSubmit.scala` (`val uninitLog = initializeLogIfNecessary(true, silent = true)`)  -> `Logging.scala` (`val replLevel = Option(replLogger.getLevel()).getOrElse(Level.WARN)`), the log level for `rootLogger ` is overiri to `WARN ` but the status of the driver and kill driver commands status have been logging with `INFO ` log level, so there is nothing printing on the console for status and kill commands. This PR overrides the `logInfo()` for `RestSubmissionClient` and redirects the msg to `printStream`.

## How was this patch tested?

I verified it manually by running status and kill commands, these are the results with and without the PR change.

- Without the PR change

```
[user1@user1-work-pc bin]$ ./spark-submit --master spark://user1-work-pc:6066 --status driver-20180803165641-0000
[user1@user1-work-pc bin]$ ./spark-submit --master spark://user1-work-pc:6066 --kill driver-20180803165641-0000
```

- With the PR change

```
[user1@user1-work-pc bin]$ ./spark-submit --master spark://user1-work-pc:6066 --kill driver-20180803165641-0000
Submitting a request to kill submission driver-20180803165641-0000 in spark://user1-work-pc:6066.
Server responded with KillSubmissionResponse:
{
  "action" : "KillSubmissionResponse",
  "message" : "Driver driver-20180803165641-0000 has already finished or does not exist",
  "serverSparkVersion" : "2.4.0-SNAPSHOT",
  "submissionId" : "driver-20180803165641-0000",
  "success" : false
}

[user1@user1-work-pc bin]$ ./spark-submit --master spark://user1-work-pc:6066 --status driver-20180803165641-0000
Submitting a request for the status of submission driver-20180803165641-0000 in spark://user1-work-pc:6066.
Server responded with SubmissionStatusResponse:
{
  "action" : "SubmissionStatusResponse",
  "driverState" : "FINISHED",
  "serverSparkVersion" : "2.4.0-SNAPSHOT",
  "submissionId" : "driver-20180803165641-0000",
  "success" : true,
  "workerHostPort" : "xx.x.xx.xxx:42040",
  "workerId" : "worker-20180803165615-10.3.66.149-42040"
}
```

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala [hunk 1]
```diff
@@ -98,17 +98,24 @@ private[spark] class SparkSubmit extends Logging {
...
-      .requestSubmissionStatus(args.submissionToRequestStatusFor)
+    createRestSubmissionClient(args).requestSubmissionStatus(args.submissionToRequestStatusFor)
+  }
+
+  /**
+   * Creates RestSubmissionClient with overridden logInfo()
+   */
+  private def createRestSubmissionClient(args: SparkSubmitArguments): RestSubmissionClient = {
+    new RestSubmissionClient(args.master) {
+      override protected def logInfo(msg: => String): Unit = printMessage(msg)
```
Comment (felixcheung) [inline core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala]: doesn't this change the behavior even when the logger is initialized/working?

**Review by felixcheung — COMMENTED** [review]
I'm not sure how the PR title is related to the change here?

**Comment by devaraj-kavali** [comment]
> I'm not sure how the PR title is related to the change here?

As an user perspective, when they don't see any o/p for status/kill commands, they would probably assume that not working but the reason is that it is not showing the command response message to the user. I used the same JIRA title as the PR title here.

File: core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala (hunk 1 above)
Comment (devaraj-kavali) [reply core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala]: When `isInterpreter = true`(repl shell) the log is initialized and working but the log level is setting to WARN due to that `RestSubmissionClient` logInfo messages are not showing which are as part of the response. This PR change effects when `isInterpreter = true` and for status/kill commands, and doesn't change the other behavior.

**Comment by felixcheung** [comment]
I think we generally describe the change in PR title. what user see you can put as JIRA title.

File: core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala (hunk 1 above)
Comment (felixcheung) [reply core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala]: this is not necessarily always the case - user can config log level easily?

Comment (devaraj-kavali) [reply core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala]: I agree, user can change this log level. If the users configure the log level as WARN or above(WARN is the default log level) then they can't see any update/status from status and kill commands. I think we cannot expect the user to configure the log level to INFO if they want to run status and kill commands with status/update. Please let me know if you have any thoughts to fix this better, I can make the changes. Thanks

**Comment by devaraj-kavali** [comment]
@felixcheung can you check this PR, please let me know if there is anything need to be updated.

**Review by srowen — COMMENTED** [review]
No, we shouldn't hack what log levels mean and do here. If the messages are very important, we can log at 'warn'. Or else, selectively enable info logging for just this package or logger.

**Pull request closed without merging**